From 189f85a9f7503ff63dad8ff0c7d6f9164e170829 Mon Sep 17 00:00:00 2001
From: Nannan Zhao <nannanzhao@nannans-mbp.almaden.ibm.com>
Date: Wed, 28 Jun 2017 13:59:02 -0700
Subject: [PATCH] fixing undefined10

---
 registry/manifest.go | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/registry/manifest.go b/registry/manifest.go
index 4f8daba..7b13a9b 100644
--- a/registry/manifest.go
+++ b/registry/manifest.go
@@ -49,7 +49,14 @@ func (registry *Registry) ManifestV2(repository, reference string) (*http.Respon
 		return nil, err
 	}
 
-	req.Header.Set("Accept", manifestV2.MediaTypeManifest)
+	//for _, t := range distribution.ManifestMediaTypes() {
+	//	req.Header.Add("Accept", t)
+	//}
+
+	req.Header.Add("Accept", manifestV2.MediaTypeManifest)
+	req.Header.Add("Accept", manifestV1.MediaTypeManifest) //application/vnd.docker.distribution.manifest.list.v2+json
+	req.Header.Add("Accept", "application/vnd.docker.distribution.manifest.list.v2+json")
+
 	resp, err := registry.Client.Do(req)
 	if err != nil {
 		return nil, err
-- 
2.11.0 (Apple Git-81)

